runtime/Go/antlr/lexer.go

```diff
@@ -25,11 +27,12 @@ type Lexer interface {
 type BaseLexer struct {
 	*BaseRecognizer
 
-	Interpreter         *LexerATNSimulator
+	Interpreter         ILexerATNSimulator
 	TokenStartCharIndex int
 	TokenStartLine      int
 	TokenStartColumn    int
 	ActionType          int
+	Virt                Lexer // The most derived lexer implementation. Allows virtual method calls.
```

Comment:
Hmm. Let me give some context. `*PositionAdjustingLexer` overrides `NextToken` and then calls `*BaseLexer`'s implementation of the same. Previously, `*BaseLexer` had calls in `NextToken` to `*BaseLexer.Emit`, which `*PositionAdjustingLexer` also overrides. This method call was _not_ to an interface type, hence no dynamic dispatch and the wrong method was called (that of the "super class").

By passing the interface type `Lexer` (runtime type of `*PositionAdjustLexer`) to the "super class" (`*BaseLexer`) it can make proper virtual method calls. I don't think your solution would do this.